tap_github/tests/test_authenticator.py

```diff
@@ -112,3 +118,309 @@ def test_has_calls_remaining_fails_if_few_calls_remaining_and_reset_time_not_rea
         token_manager.update_rate_limit(mock_response_headers)
 
         assert not token_manager.has_calls_remaining()
+
+
+class TestAppTokenManager:
+
+    def test_initialization_with_3_part_env_key(self):
+        with patch.object(AppTokenManager, "claim_token", return_value=None):
+            token_manager = AppTokenManager("12345;;key\\ncontent;;67890")
+            assert token_manager.github_app_id == "12345"
+            assert token_manager.github_private_key == "key\ncontent"
+            assert token_manager.github_installation_id == "67890"
+
+    def test_initialization_with_2_part_env_key(self):
+        with patch.object(AppTokenManager, "claim_token", return_value=None):
+            token_manager = AppTokenManager("12345;;key\\ncontent")
+            assert token_manager.github_app_id == "12345"
+            assert token_manager.github_private_key == "key\ncontent"
+            assert token_manager.github_installation_id == ""
+
+    def test_initialization_with_malformed_env_key(self):
+        with pytest.raises(ValueError) as exc_info:
+            AppTokenManager("12345key\\ncontent")
+
+        assert str(exc_info.value) == (
+            "GITHUB_APP_PRIVATE_KEY could not be parsed. The expected format is "
+            '":app_id:;;-----BEGIN RSA PRIVATE KEY-----\\n_YOUR_P_KEY_\\n-----END RSA PRIVATE KEY-----"'
+        )
```

Comment:
Would using [`pytest.raises(ValueError, match=...)`](https://docs.pytest.org/en/latest/reference/reference.html#pytest.raises) be preferred here?

Reply:
Good call!  Pushed a change for this. :)